orbitx/flight_gui.py

```diff
@@ -125,6 +90,260 @@ def callback(*_):
+             lambda: f"{self._h_speed(self._reference.name):,.7g} m/s ",
+             "Horizontal speed of habitat across reference surface",
+             False),
+            ("Targ alt",
+             lambda: f"{self._altitude(self._target.name):,.7g} m",
+             "Altitude of habitat above reference surface",
+             True),
+            ("Targ speed",
+             lambda: f"{self._speed(self._target.name):,.7g} m/s",
+             "Altitude of habitat above reference surface",
+             False),
+            ("Throttle",
+             lambda: f"{self._habitat.throttle:.1%}",
+             "Percentage of habitat's maximum rated engines",
+             True)
+        ]:
+            self._wtexts.append(self._vpython.wtext(text=text_gen_func()))
+            self._wtexts[-1].text_func = text_gen_func
+            self._scene.caption += f"""<tr>
+                <td {"class='newsection'" if new_section else ""}>
+                    {caption}
+                </td>
+                <td class="num{" newsection" if new_section else ""}">
+                    <div id="{div_id}">{self._wtexts[-1].text}</div>
+                </td>
+                <td class="helptext{" newsection" if new_section else ""}">
+                    {helptext}
+                </td>
+                </tr>\n"""
+            div_id += 1
+        self._scene.caption += "</table>"
+
+        #self._scene.append_to_caption("Acc: XXX" + "\n")
+        #self._scene.append_to_caption("Vcen: XXX" + "\n")
+        #self._scene.append_to_caption("Vtan: XXX" + "\n")
+        # self._scene.append_to_caption("\n")
+        self._set_menus()
+        self._scene.append_to_caption(style.HELP_CHECKBOX)
+        self._scene.append_to_caption(" Help text")
+        self._scene.append_to_caption(style.INPUT_CHEATSHEET)
+
+        self._scene.append_to_caption(style.VPYTHON_CSS)
+        self._scene.append_to_caption(style.VPYTHON_JS)
+    # end of _set_caption    
+
+    def recentre_camera(self, planet_name: str) -> None :
+        """Change camera to focus on different object
+
+        Because GPU(Graphics Processing Unit) cannot deal with extreme case of
+        scene center being approximately 1e11 (planets position), origin
+        entity should be reset every time when making a scene.center update.
+        """
+        try:
+            if planet_name == "Sun":
+                self._scene.range = self._spheres["Sun"].radius * 15000
+            else:
+                self._scene.range = self._spheres[planet_name].radius * 2
+
+            self._scene.camera.follow(self._spheres[planet_name])
+            self._set_origin(planet_name)
+
+        except KeyError:
+            log.error(f'Unrecognized planet to follow: "{planet_name}"')
+        except IndexError:
+            log.error(f'Unrecognized planet to follow: "{planet_name}"')
+    # end of recentre_camera
+
+    #################################################################################
```

Comment:
what is this line for?

Reply:
just for eye candy now. It's hard to read code in Flight_Gui because it's too long. It should be deleted later.

Reply:
Ok, good to know thank you